subprojects/core/src/main/java/org/gradle/api/internal/file/collections/jdk7/Jdk7DirectoryWalker.java

```diff
@@ -91,17 +92,30 @@ private FileVisitDetails getFileVisitDetails(Path file, BasicFileAttributes attr
                     File child = file.toFile();
                     FileVisitDetails dirDetails = directoryDetailsHolder.peek();
                     RelativePath childPath = dirDetails != null ? dirDetails.getRelativePath().append(!isDirectory, child.getName()) : rootPath;
-                    return new DefaultFileVisitDetails(child, childPath, stopFlag, fileSystem, fileSystem, isDirectory, attrs.lastModifiedTime().toMillis(), attrs.size());
+                    if (attrs == null) {
+                        return new UnauthorizedFileVisitDetails(child, childPath);
```

Comment:
I don't think this is actually the correct behaviour for all the consumers of this code. We've chosen an implementation that works for one very specific case, but in doing so I think we've broken, or at least changed a whole bunch of other cases, because we're now ignoring a failure but we don't know whether or not the consumer cares about this failure or not.